torch/nn/modules/container.py

```diff
@@ -606,6 +606,20 @@ def __iter__(self) -> Iterator[str]:
     def __contains__(self, key: str) -> bool:
         return key in self._parameters
 
+    def setdefault(self, key: str, default: Optional['Parameter'] = None) -> 'Parameter':
+        """If key is in the ParameterDict, return its parameter.
+        If not, insert `key` with a parameter `default` and return `default`.
+        `default` defaults to `None`.
+
+        Args:
+            key (string): key to set default for
+            default (:class:`~torch.nn.Parameter`): the parameter set to the key
+        """
+        if key in self._parameters:
+            return self._parameters[key]
+        self[key] = default
```

Comment:
It seems that `mypy` is detecting an error on this line because you're assigning `default` to `self[key]`, which accepts a `Parameter`, whereas `default` is of type `Optional[Parameter]`. I'm not sure how to resolve this since this should be a reasonable assignment. One way (although not that clean) would be to just tell `mypy` to ignore that line:
```
# type: ignore
```

Reply:
Sorry I overlooked this one before merging.

I just created a new PR that adds `# type: ignore` as you suggested. See https://github.com/thisisalbertliang/pytorch/pull/8